### ISSUE TITLE
i have an opposite conclusion about norm and quality.

### ISSUE THREAD

**Issue by luckydog11** [mk-minchul/AdaFace#6]
hi, I'm trying to use your adaface_loss in my code, but I find a problem about that the feature with low norm has high quality, and I mean i have an opposite conclusion about norm and quality.

**Comment by mk-minchul** [comment]
Hi luckydog11. That is a strange behavior of a model. But if that phenomenon is persistent in your model and you would like to apply AdaFace, you can change the sign of the hyper parameter 'h'. Then the margin ordering will be reversed.

**Comment by luckydog11** [comment]
Thank you, mk-minchul, I have a question if the relationship between the norm and quality is universal. I train IR50 using MS1MV3 and others with adaface, but I find that i still have an opposite conclusion about norm and quality. For "a strange behavior of a model", what are you meaning for? Or what are the preconditions for such a relationship? 

best regards,

发自我的iPhone

**Comment by mk-minchul** [comment]
I believe that behavior is somewhat common as I observed in my trained models and this paper https://arxiv.org/abs/1703.09507 also mentions it. The high qual vs high norm relationship will be likely to appear with the softmax loss function as it is easier for the confident samples to maximize the value corresponding to the GT label, to minimize the loss. I think it is worth investigating why some models do not exhibit that behavior. My initial guess would be it is related to the model size, dataset size and the possibly the margin. Hope that helps.

**Comment by NNDam** [comment]
I got similar problem when analyst our Cosface model (custom dataset)
![image](https://user-images.githubusercontent.com/40813954/183546982-1ea84651-e16e-46d4-9634-579b6f16dc2a.png)
